pkg/monitor/cluster/clusterobjectconditions.go

```diff
@@ -13,30 +13,45 @@ import (
 	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 )
 
-var aroOperatorConditionsExpected = map[string]operatorv1.ConditionStatus{
+var clusterObjectConditionsExpected = map[string]operatorv1.ConditionStatus{
+	// ARO Operator Conditions
 	arov1alpha1.InternetReachableFromMaster: operatorv1.ConditionTrue,
 	arov1alpha1.InternetReachableFromWorker: operatorv1.ConditionTrue,
+	// Service Principal Condition
+	arov1alpha1.ServicePrincipalValid: operatorv1.ConditionTrue,
 }
 
-func (mon *Monitor) emitAroOperatorConditions(ctx context.Context) error {
+var metricname = map[string]string{
```

Comment:
instead of having two separate maps to add the same key to, can we have a struct to contain the ExpectedStatus and the MetricName for each Condition?

```golang
type conditionMetric struct {
	expectedStatus operatorv1.ConditionStatus
	metricName     string
}

var clusterObjectConditionMetrics = map[string]conditionMetric{
	arov1alpha1.InternetReachableFromMaster: {
		expectedStatus: operatorv1.ConditionTrue,
		metricName: "arooperator.conditions",
	},
}
```

Reply:
Made the suggested changes.